### PR TITLE
Preliminary support for parsing OpenQASM 3.0

### PR DESCRIPTION
**Pull request by dstrain115**
- This adjusts the lexer and parser to handle programs from OpenQASM 3.0.

### CODE REVIEW

File: cirq-core/cirq/contrib/qasm_import/_parser.py [hunk 1]
```diff
@@ -315,13 +322,19 @@ def p_circuit_empty(self, p):
 
     def p_new_reg(self, p):
         """new_reg : QREG ID '[' NATURAL_NUMBER ']' ';'
-        | CREG ID '[' NATURAL_NUMBER ']' ';'"""
-        name, length = p[2], p[4]
+        | QUBIT '[' NATURAL_NUMBER ']' ID ';'
```
Comment (pavoljuhas) [inline cirq-core/cirq/contrib/qasm_import/_parser.py:325]: The OpenQASM 3 seems to allow [scalar qubits](https://openqasm.com/versions/3.0/language/types.html#qubits) defined as `qubit some_name;`.

Would it be feasible to handle it here?

**Review by pavoljuhas — APPROVED** [review]
LGTM.  Please see the inline question.

File: cirq-core/cirq/contrib/qasm_import/_parser.py (hunk 1 above)
Comment (dstrain115) [reply cirq-core/cirq/contrib/qasm_import/_parser.py:325]: Added scalar qubits (and bits)

File: cirq-core/cirq/contrib/qasm_import/_parser.py [hunk 2]
```diff
@@ -315,13 +322,28 @@ def p_circuit_empty(self, p):
...
+        | CREG ID '[' NATURAL_NUMBER ']' ';'
+        | BIT '[' NATURAL_NUMBER ']' ID ';'
+        | BIT ID ';'
+        """
+        if p[1] == "qreg" or p[1] == "creg":
+            # QREG ID '[' NATURAL_NUMBER ']' ';'
+            name, length = p[2], p[4]
+        else:
+            if len(p) < 5:
+                # QUBIT ID ';'
```
Comment (pavoljuhas) [inline cirq-core/cirq/contrib/qasm_import/_parser.py:336]:
```suggestion
                # QUBIT ID ';' | BIT ID ';'
```

**Review by pavoljuhas — APPROVED** [review]

**Pull request merged**